openmsupply/application-manager-web-app
# #711 & #707 Review changes requested progress

Fixes #711 & fixes #707

## Back-end PR to be used:
[#711F](https://github.com/openmsupply/application-manager-server/pull/371)

### Changes 
- #707 had the action to create duplicated reviews from previous review already implemented. The additions done here was only on `ReviewSectionRowAction` to correctly display the `action` **Update** and allow the Review to be started
- Added new fragment `consolidatorResponsesFragment` and added to `getReviewResponsesQuery` to get the connected nodes using `reviewResponseLinkId` on each returned `thisReviewResponse`
- Renamed some files to get a similar way to construct progress for each users...
- The new structure changes has now basically added:
```
{
  sortedPages: [
    {
      state: [
        {
           thisReviewLastesReponse: {
            reviewResponsesByReviewResponseLinkId: {
              nodes: [...]: ReviewResponse[]
            }
         },
         thisReviewPreviousReponse: {
            reviewResponsesByReviewResponseLinkId: {
              nodes: [...]: ReviewResponse[]
            }
          },
          changeRequestsProgress: {
           totalChangeRequests: number,
           doneChangeRequests: number
          },
          ...
       ]: PageElement
      },
    },
  ]: Page,
  sortedSections: [
    changeRequestsProgress: {
      totalChangeRequests: number,
      doneChangeRequests: number
    },
  ],
  isChangeRequest: boolean
}
``` 

### Test

- Use application serial: 34567 to test. It has CHANGES_REQUESTED from **Consolidator2** to **Reviewer1** on Section 1. 
- Click on the review HomePage on the **Update** action and check out the progress and validation considering those. 
- The progress at the start is considering both: responses reviewed + updated changes requested
- It's not visible on the Review page yet, but only the 2 first questions have changes requests, so changing those 2 should update the progress to **Completed** and allow the review to be submitted.
